typescript-closures/my-app/src/Counter.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+type Data = {
+  foo: string;
+};
+
+export function Counter() {
+  const [data, setData] = useState<Data>();
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
+    function getData() {
+      // fetch data
+      return { foo: 'bar' };
+    }
+    const data = getData();
+    setData(data);
+    setCounter((prev) => prev + 1);
+  }, [setData]);
```

Comment:
`setData` is a state setter, so it does not need to be put in a dependency array.

You fixed the warning by moving getData into the useEffect. Another way to fix it is to use `useCallback` when creating the `getData` function.